x-pack/plugins/fleet/public/applications/fleet/mock/epm/details_api.ts

```diff
@@ -0,0 +1,539 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import {
+  composeHttpHandlerMocks,
+  httpHandlerMockFactory,
+  ResponseProvidersInterface,
+} from '../http_handler_mock_factory';
+import {
+  agentPolicyRouteService,
+  epmRouteService,
+  GetAgentPoliciesResponse,
+  GetFleetStatusResponse,
+  GetInfoResponse,
+  GetPackagePoliciesResponse,
+  GetStatsResponse,
+  KibanaAssetType,
+  packagePolicyRouteService,
+} from '../../../../../common';
+import {
+  agentsSetupApiMock,
+  AgentsSetupResponseProvidersMock,
+  fleetSetupApiMock,
+  FleetSetupResponseProvidersMock,
+} from '../setup';
+
+export const epmPackageResponse = (): GetInfoResponse =>
+  // @ts-ignore
```

Comment:
Can we either a) fix whatever is causing the TS error b) add a note about the error so we know when/if ignore can be removed

Reply:
I attempted to fix this, but am still getting an error which I am not sure how to fix. So I'm leaving this one in (although I changed it to `@ts-expect-error`.

Error is:
![image](https://user-images.githubusercontent.com/56442535/113351852-71e8c100-9309-11eb-83d9-7c19db10dccb.png)

I think the error is due to TS not doing mutually exclusive unions - so this type def. here seems to be the one causing the issue:

https://github.com/elastic/kibana/blob/478ad3bad5e375aa886ac750e9cbe5cc87ccbb0f/x-pack/plugins/fleet/common/types/models/epm.ts#L351-L358

These two are both used in the `GetInfoResponse`:

https://github.com/elastic/kibana/blob/478ad3bad5e375aa886ac750e9cbe5cc87ccbb0f/x-pack/plugins/fleet/common/types/models/epm.ts#L328-L330